keras_cv/models/backbones/csp_darknet/csp_darknet_backbone.py

```diff
@@ -0,0 +1,383 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""CSPDarkNet models for KerasCV. """
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.csp_darknet.csp_darknet_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.models.backbones.csp_darknet.csp_darknet_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    CrossStagePartial,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import DarknetConvBlock
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    DarknetConvBlockDepthwise,
+)
+from keras_cv.models.backbones.csp_darknet.darknet_utils import Focus
+from keras_cv.models.backbones.csp_darknet.darknet_utils import (
+    SpatialPyramidPoolingBottleneck,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class CSPDarkNetBackbone(Backbone):
+    """This class represents the CSPDarkNet architecture.
+
+    Reference:
+        - [YoloV4 Paper](https://arxiv.org/abs/1804.02767)
+        - [CSPNet Paper](https://arxiv.org/abs/1911.11929)
+        - [YoloX Paper](https://arxiv.org/abs/2107.08430)
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+    Args:
+        channels: A list of ints, the number of channels for each dark level in
+            the model.
+        depths: A list of ints, the depth for each dark level in the model.
+        include_rescaling: bool, whether to rescale the inputs. If set to
+            True, inputs will be passed through a `Rescaling(1/255.0)` layer.
+        use_depthwise: bool, whether a `DarknetConvBlockDepthwise` should be
+            used over a `DarknetConvBlock`, defaults to False.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        input_tensor: optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+
+    Returns:
+        A `keras.Model` instance.
+
+    Examples:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained backbone
+    model = keras_cv.models.CSPDarkNetBackbone.from_preset(
+        "cspdarknettiny_imagenet"
+    )
+    output = model(input_data)
+
+    # Randomly initialized backbone with a custom config
+    model = CSPDarkNetBackbone(
+        channels=[128, 256, 512, 1024],
+        depths=[3, 9, 9, 3],
+        include_rescaling=False,
+    )
+    output = model(input_data)
+    ```
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        *,
+        channels,
```

Comment:
Use `stackwise_channels` and `stackwise_depth` to match our other backbones. It also makes your unpacking below more readable.

Reply:
Done.